src/components/Card.jsx

```diff
@@ -1,54 +1,76 @@
 import { dateDiffCalc, dateFormatter } from "../util/datecalc";
-import emptyLogo from "../images/emptylogo.svg";
+import iconEmptyLogo from "../images/emptylogo.svg";
 import kebab from "../images/kebab.svg";
-import useToggle from "../hooks/useToggle";
 import star from "../images/star.svg";
-
-function SelectMenu() {
-  return (
-    <div className="selectMenu-container">
-      <div className="selectMenu">삭제하기</div>
-      <div className="selectMenu clicked">폴더에 추가</div>
-    </div>
-  );
-}
+import styles from "./Card.module.css";
+import SelectMenu from "./SelectMenu";
+import { useState } from "react";
+import DeleteLinkModal from "../modal/DeleteLinkModal/DeleteLinkModal";
+import AddToFolderModal from "../modal/AddToFolderModal/AddToFolderModal";
 
 function Card({ link }) {
-  const [kebabValue, KebabTogglefunc] = useToggle(false);
+  const [isKebabClicked, setIsKebabClicked] = useState(false);
+  const [isModalClicked, setIsModalClicked] = useState({
+    deleteLink: false,
+    addToFolder: false,
+  });
   const cardImage =
     link.imageSource || link["image_source"]
       ? link.imageSource || link["image_source"]
-      : emptyLogo;
+      : iconEmptyLogo;
   const dateOfCard = new Date(link.createdAt || link["created_at"]);
   const dateDiff = dateDiffCalc(dateOfCard);
   const formatDate = dateFormatter(dateOfCard);
 
-  const moveToUrl = () => {
+  const onClickCard = () => {
     window.open(link.url, "_blank");
   };
 
+  const onClickKebab = (e) => {
+    setIsKebabClicked((prevValue) => !prevValue);
+    e.stopPropagation();
```

Comment:
왠만하면 무조건 이벤트부터 막아주세요!
```suggestion
    e.stopPropagation();
    setIsKebabClicked((prevValue) => !prevValue);
```